[yu-sutong] iP

yu-sutong opened the pull request:
MySutong frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/yu-sutong/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yu-sutong committed:
add MySutong.java

yu-sutong committed:
edited MySutong.java for Level-1

yu-sutong committed:
Added Level-2 requirements to MySutong.java

yu-sutong committed:
Added Level-3 requirements to MySutong.java

yu-sutong committed:
Added Level-4 requirements to MySutong.java

yu-sutong committed:
Added A-TextUiTesting requirements

yu-sutong committed:
Added Level-5 requirements for MySutong.java

yu-sutong committed:
Added Level-6 requirements to MySutong.java

yu-sutong committed:
Add Level 7 save feature

yu-sutong committed:
Merge branch 'branch-Level-7'

yu-sutong committed:
Add Level 8 features

yu-sutong committed:
Merge branch 'branch-Level-8'

yu-sutong committed:
Add A-MoreOOP increments

yu-sutong committed:
Merge branch 'branch-A-MoreOOP'

yu-sutong committed:
Add A-Packages increments

yu-sutong committed:
Merge branch 'branch-A-Packages'

yu-sutong committed:
Merge branch 'add-gradle-support'

yu-sutong committed:
Add A-JUnit increments

yu-sutong committed:
Merge branch 'branch-A-JUnit'

kevin-pek commented on src/main/java/mysutong/Parser.java:
```diff
@@ -0,0 +1,92 @@
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "deadline":
+                    String[] deadlineDetails = inputs[1].split("/by");
+                    tasks.addTask(new Deadline(deadlineDetails[0].trim(), LocalDateTime.parse(deadlineDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
+                    ui.showLine();
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "event":
+                    String[] eventDetails = inputs[1].split("/from|/to");
+                    tasks.addTask(new Event(eventDetails[0].trim(), LocalDateTime.parse(eventDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm")), LocalDateTime.parse(eventDetails[2].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
+                    ui.showLine();
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "delete":
+                    int deleteIndex = Integer.parseInt(inputs[1]) - 1;
+                    Task removedTask = tasks.getTask(deleteIndex);
+                    tasks.removeTask(deleteIndex);
+                    ui.showLine();
+                    ui.showMessage("Noted. I've removed this task:");
+                    ui.showMessage(removedTask.toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                default:
+                    throw new UnknownCommandException("I'm sorry, but I don't know what that means.");
+            }
```
```suggestion
            switch (command) {
            case "bye":
                ui.showGoodbye();
                System.exit(0);
                break;
            case "list":
                ui.showTaskList(tasks);
                break;
            case "mark":
                int markIndex = Integer.parseInt(inputs[1]) - 1;
                tasks.getTask(markIndex).markAsDone();
                ui.showLine();
                ui.showMessage("Nice! I've marked this task as done:");
                ui.showMessage(tasks.getTask(markIndex).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "unmark":
                int unmarkIndex = Integer.parseInt(inputs[1]) - 1;
                tasks.getTask(unmarkIndex).unmark();
                ui.showLine();
                ui.showMessage("OK, I've marked this task as not done yet:");
                ui.showMessage(tasks.getTask(unmarkIndex).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "todo":
                tasks.addTask(new Todo(inputs[1]));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "deadline":
                String[] deadlineDetails = inputs[1].split("/by");
                tasks.addTask(new Deadline(deadlineDetails[0].trim(), LocalDateTime.parse(deadlineDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "event":
                String[] eventDetails = inputs[1].split("/from|/to");
                tasks.addTask(new Event(eventDetails[0].trim(), LocalDateTime.parse(eventDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm")), LocalDateTime.parse(eventDetails[2].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "delete":
                int deleteIndex = Integer.parseInt(inputs[1]) - 1;
                Task removedTask = tasks.getTask(deleteIndex);
                tasks.removeTask(deleteIndex);
                ui.showLine();
                ui.showMessage("Noted. I've removed this task:");
                ui.showMessage(removedTask.toString());
                ui.showLine();
                storage.save(tasks);
                break;
            default:
                throw new UnknownCommandException("I'm sorry, but I don't know what that means.");
          }
```

The coding standard mentions that `switch` statements should not have indentation for `case` clauses. Default formatting for your IDE may have conflicted with this.

kevin-pek commented on src/main/java/mysutong/Storage.java:
```diff
@@ -0,0 +1,66 @@
+            String description = parts[2];
+
+            Task task;
+            switch (taskType) {
+                case "T":
+                    task = new Todo(description);
+                    break;
+                case "D":
+                    LocalDateTime deadlineDate = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("d/M/yyyy HHmm"));
+                    task = new Deadline(description, deadlineDate);
+                    break;
+                case "E":
+                    LocalDateTime eventStart = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("d/M/yyyy HHmm"));
+                    LocalDateTime eventEnd = LocalDateTime.parse(parts[4], DateTimeFormatter.ofPattern("d/M/yyyy HHmm"));
+                    task = new Event(description, eventStart, eventEnd);
+                    break;
+                default:
+                    throw new IllegalStateException("Unknown task type: " + taskType);
+            }
```
Same issue with switch formatting.

kevin-pek commented on src/main/java/mysutong/Storage.java:
```diff
@@ -0,0 +1,66 @@
+import mysutong.Deadline;
+import mysutong.Event;
+
+import java.io.*;
```
Imports should be made explicit according to the coding standard.
```suggestion
import java.io.File;
import java.io.FileNotFoundException;
import java.io.BufferedWriter;
import java.io.IOException;
```

kevin-pek commented on src/main/java/mysutong/Parser.java:
```diff
@@ -0,0 +1,92 @@
+
+                case "event":
+                    String[] eventDetails = inputs[1].split("/from|/to");
+                    tasks.addTask(new Event(eventDetails[0].trim(), LocalDateTime.parse(eventDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm")), LocalDateTime.parse(eventDetails[2].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
```
This line is past the 120 character line limit for the coding standards. It would be good to consider doing line wrapping. Alternatively you might want to separate the logic into multiple lines instead.

```suggestion
                    tasks.addTask(new Event(eventDetails[0].trim(),
                            LocalDateTime.parse(eventDetails[1].trim(),
                                    DateTimeFormatter.ofPattern("d/M/yyyy HHmm")),
                            LocalDateTime.parse(eventDetails[2].trim(),
                                    DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
```

kevin-pek reviewed (commented):
LGTM, just have some nitpicks with regards to the coding standards. Have highlighted them for your consideration. Thanks!

yu-sutong committed:
Add A-Jar increments

yu-sutong committed:
Merge branch 'branch-A-Jar'

yu-sutong committed:
Make the code follow OOP

yu-sutong committed:
Add A-JavaDoc increments

yu-sutong committed:
Add A-CodingStandard increment

yu-sutong committed:
Add Level-9 increments

yu-sutong committed:
Merge branch 'branch-A-JavaDoc'

yu-sutong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/mysutong/Storage.java

yu-sutong committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/mysutong/Parser.java
#	src/main/java/mysutong/Ui.java

yu-sutong committed:
Add JavaDoc for TaskList methods

yu-sutong committed:
Make it more OOP

yu-sutong committed:
Add Level 10 increments

yu-sutong committed:
Make it more OOP

yu-sutong committed:
Add JavaDoc to classes

yu-sutong committed:
Add A-CheckStyle requirement

yu-sutong committed:
Add assertion to ensure valid command input

The executeCommand method in Parser handles user commands by splitting the input string into two parts: the command itself and any additional arguments.

Currently, the split operation does not validate that the input string contains at least one element, which could potentially lead to unexpected behaviour or bugs if empty input is passed.

To mitigate this, let's add an assert statement to ensure that the inputs array has at least one element. This ensures that the code always receives a valid command to process. In situations where assertions are enabled, this will help catch empty or invalid commands during development and testing.

While this doesn't address user input validation directly (since assert can be disabled in production), it adds an extra layer of safety during development.

This change also avoids unnecessary array access errors and improves the stability of the command handling logic.

yu-sutong committed:
Refactor code style to adhere to Checkstyle guidelines

This commit refactors the code to conform to the Checkstyle configuration, ensuring consistency and compliance with coding standards.

Changes made:
- Replaced wildcard imports (e.g., import static org.junit.jupiter.api.Assertions.*) with explicit imports for better clarity and to avoid unnecessary imports.
- Corrected import order to follow the standard conventions, ensuring that java.* imports are grouped together and listed alphabetically.
- Fixed spacing issues, such as ensuring a single space between comments and code, and after operators (e.g., =, +).
- Improved overall code readability and conformance to Checkstyle rules by addressing minor formatting inconsistencies.

These changes are intended to improve the maintainability of the codebase by ensuring that it follows consistent styling rules as enforced by Checkstyle. By doing this, we minimize the chances of formatting-related errors and enhance code readability across the project.

All changes are purely stylistic and do not alter the functionality of the code.

yu-sutong committed:
Merge pull request #2 from yu-sutong/branch-A-Assertions

Add assertion to ensure valid command input

yu-sutong committed:
Merge pull request #3 from yu-sutong/branch-A-CodeQuality

Refactor code style to adhere to Checkstyle guidelines

yu-sutong committed:
Implement C-Priority

yu-sutong committed:
Fix some logical issues in TaskList

yu-sutong committed:
Add A-UserGuide requirements

yu-sutong committed:
Edit README.md
